geps/gep-3614/index.md

```diff
@@ -0,0 +1,71 @@
+# GEP-3614: Firewall
+
+* Issue: [#3614](https://github.com/kubernetes-sigs/gateway-api/issues/3614)
+* Status: Provisional
+
+## TLDR
+
+The ability to attach firewall rules for ingress L3, L4 and L7 Gateway traffic.
+
+## Motivation
+
+`Gateways` are commonly exposed to the internet, which puts them as risk of
+attack. Internal networks can become compromised as well. We should provide
+tooling, documentation and best-practices for users to restrict and control
+access to their `Gateways`.
+
+### User Stories
+
+* As an application developer, I want to allow specific IPs to access my
+  application.
+* As an application developer, I want all ingress traffic for my Gateways to be
+  restricted to my CDN.
+* As an application developer, I want to block all requests from specific user
+  agents OR only allow specific user agents.
+* As a gateway operator I want to be able to identify and block malformed HTTP
+  requests before they reach backend applications.
+* As a gateway operator I want to be able to provide my own signature-based
+  detection rulesets to spot patterns of known malicious traffic and block
+  them, updating those rules dynamically over time.
+* As a gateway operator I want to attach complete rulesets maintained by
+  upstream standards bodies to block well known common threats and dynamically
+  update for new threats over time.
+* As a cluster operator I want to be able to block traffic to gateways from
+  specific geographical regions, or only allow specific regions.
+
```

Comment:
* As a gateway operator I want to be able to block traffic to gateways based on custom logic provided as portable scripts e.g. https://github.com/jcchavezs/coraza-exec-action-wasm